productivity-suite/app/legacy-app/src/components/pages/TodoLists.tsx

```diff
@@ -1,67 +1,59 @@
-import { useEffect, useState } from "react";
-import { useNavigate } from "react-router-dom";
-import { TodoList } from "shared";
+import { getBus } from "piercing-library";
+import { useEffect, useRef, useState } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
 import "./TodoLists.css";
 
 export function TodoLists() {
+  const ref = useRef<HTMLDivElement>(null);
   const [selectedListName, setSelectedListName] = useState<string | null>(null);
 
-  const [todoEnteringAnimation, setTodoEnteringAnimation] = useState<
-    "previous" | "next" | undefined
-  >();
-  const [showTodos, setShowTodos] = useState<boolean>(true);
+  const location = useLocation();
 
   useEffect(() => {
-    const match = /\/todos\/([^/]+)/.exec(window.location.pathname);
-    if (match) {
-      setSelectedListName(decodeURIComponent(match[1]));
+    const pathname = location.pathname;
```

Comment:
@petebacondarwin I put back this change, what was I thinking 🤦 , you definitely (unfortunately) need to get the list name from the location because the one in the bus will not get updated 😕 
(I made that change earlier today so I think I flat out broke it but didn't notice it 😓 )